webui/react/src/pages/Cluster/utils.ts

```diff
@@ -17,18 +19,34 @@ export interface ResourceAllocationChartSeries {
 export const mapResourceAllocationApiToChartSeries = (
   apiRes: Array<V1ResourceAllocationAggregatedEntry>,
   grouping: GroupBy,
+  users: DetailedUser[],
 ): ResourceAllocationChartSeries => {
   return {
     groupedBy: grouping,
     hoursByAgentLabel: mapToChartSeries(apiRes.map(item => item.byAgentLabel)),
     hoursByExperimentLabel: mapToChartSeries(apiRes.map(item => item.byExperimentLabel)),
     hoursByResourcePool: mapToChartSeries(apiRes.map(item => item.byResourcePool)),
-    hoursByUsername: mapToChartSeries(apiRes.map(item => item.byUsername)),
+    hoursByUsername: mapToChartSeries(apiRes.map(item => {
+      return mapPeriodToDisplayNames(item.byUsername, users);
+    })),
     hoursTotal: { total: apiRes.map(item => secondToHour(item.seconds)) },
     time: apiRes.map(item => item.periodStart),
   };
 };
 
+const mapPeriodToDisplayNames = (
+  period: Record<string, number>,
+  users: DetailedUser[],
+): Record<string, number> => {
+  const result:Record<string, number> = {};
+  Object.keys(period).forEach(key => {
+    const user = users.find(u => u.username === key);
+    const displayName = getDisplayName(user);
+    result[displayName] = period[key];
+  });
+  return result;
```

Comment:
nit: you could do this as a reduce, something like
```
return (Object.keys(period).reduce((result, key) => {
    result[getDisplayName(users.find(u => u.username === key)] = period[key];
    return result;
}, {}));
```
But the way you have it works too, and might be easier to read.